pkg/encoding/sanify.go

```diff
@@ -0,0 +1,49 @@
+package encoding
+
+import (
+	"github.com/pkg/errors"
+	"github.com/surajssd/kapp/pkg/spec"
+)
+
+func fixApp(app *spec.App) error {
+
+	// fix app.Services
+	for i, service := range app.Services {
+		if err := fixService(&service, app); err != nil {
+			return errors.Wrapf(err, "Unable to fix service: %v", service.Name)
+		}
+		app.Services[i] = service
+	}
+
+	// fix app.PersistentVolumes
+	for i, pVolume := range app.PersistentVolumes {
+		if err := fixPersistentVolume(&pVolume, app); err != nil {
+			return errors.Wrapf(err, "Unable to fix persistentVolume: %v", pVolume.Name)
+		}
+		app.PersistentVolumes[i] = pVolume
+	}
+
+	return nil
+}
+
+func fixService(service *spec.ServiceSpecMod, app *spec.App) error {
+	if service.Name == "" {
+		if len(app.Services) == 1 {
+			service.Name = app.Name
+		} else {
+			return errors.New("More than one service mentioned, please specify name for each one")
+		}
+	}
+	return nil
+}
+
+func fixPersistentVolume(pVolume *spec.PersistentVolume, app *spec.App) error {
```

Comment:
the same thing, this is a per persistentVolume function, and we need app just to get the length.
I don't see why should we be iterating over app.PersistentVolumes again